### PR TITLE
Port/more of colton

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Port colton prs:
- Add itemIcons to builder request
- Fix concurrency issue

Review please

### CODE REVIEW

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 28 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 27 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TileEntityWareHouse.java#L159](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java#L159): Refactor this method to reduce its Cognitive Complexity from 26 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [baker](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker): Add a 'package-info.java' file to document the 'baker' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 27 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 26 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TileEntityWareHouse.java#L159](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java#L159): Refactor this method to reduce its Cognitive Complexity from 26 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [baker](https://github.com/Minecolonies/minecolonies/blob/da51cb4aa46faa54f36418983a7fcc338c70af9c/src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker): Add a 'package-info.java' file to document the 'baker' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 27 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 26 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TileEntityWareHouse.java#L159](https://github.com/Minecolonies/minecolonies/blob/3f8d11e1cc0251b0b6ddfa0d2df641e3bb14be9c/src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java#L159): Refactor this method to reduce its Cognitive Complexity from 26 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/3f8d11e1cc0251b0b6ddfa0d2df641e3bb14be9c/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/3f8d11e1cc0251b0b6ddfa0d2df641e3bb14be9c/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/3f8d11e1cc0251b0b6ddfa0d2df641e3bb14be9c/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/3f8d11e1cc0251b0b6ddfa0d2df641e3bb14be9c/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/3f8d11e1cc0251b0b6ddfa0d2df641e3bb14be9c/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/3f8d11e1cc0251b0b6ddfa0d2df641e3bb14be9c/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/3f8d11e1cc0251b0b6ddfa0d2df641e3bb14be9c/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/3f8d11e1cc0251b0b6ddfa0d2df641e3bb14be9c/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [baker](https://github.com/Minecolonies/minecolonies/blob/3f8d11e1cc0251b0b6ddfa0d2df641e3bb14be9c/src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker): Add a 'package-info.java' file to document the 'baker' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Comment by Raycoms** [comment]
@xavierh

File: src/main/java/com/minecolonies/blockout/controls/ItemIcon.java [hunk 1]
```diff
@@ -52,8 +57,33 @@ public void setItem(final ItemStack itemStack)
...
+            RenderHelper.enableGUIStandardItemLighting();
+            drawItemStack(itemStack, x, y);
+            GlStateManager.popMatrix();
+        }
+    }
+
+    /**
+     * Modified from GuiContainer
+     */
+    private void drawItemStack(ItemStack stack, int x, int y)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/blockout/controls/ItemIcon.java:72]: Will rain a final task all over the repo soon anyway.

**Comment by marvin-bitterlich** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1184/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 27 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 26 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TileEntityWareHouse.java#L159](https://github.com/Minecolonies/minecolonies/blob/84d9e2458d730b73f54db09b67f5175c82552f37/src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java#L159): Refactor this method to reduce its Cognitive Complexity from 26 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/84d9e2458d730b73f54db09b67f5175c82552f37/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/84d9e2458d730b73f54db09b67f5175c82552f37/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/84d9e2458d730b73f54db09b67f5175c82552f37/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/84d9e2458d730b73f54db09b67f5175c82552f37/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/84d9e2458d730b73f54db09b67f5175c82552f37/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/84d9e2458d730b73f54db09b67f5175c82552f37/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/84d9e2458d730b73f54db09b67f5175c82552f37/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/84d9e2458d730b73f54db09b67f5175c82552f37/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [baker](https://github.com/Minecolonies/minecolonies/blob/84d9e2458d730b73f54db09b67f5175c82552f37/src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker): Add a 'package-info.java' file to document the 'baker' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 27 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 26 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [TileEntityWareHouse.java#L159](https://github.com/Minecolonies/minecolonies/blob/a4011eb9aa8ae6e516f83f9d4517dedbdeea83f1/src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java#L159): Refactor this method to reduce its Cognitive Complexity from 26 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/a4011eb9aa8ae6e516f83f9d4517dedbdeea83f1/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/a4011eb9aa8ae6e516f83f9d4517dedbdeea83f1/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/a4011eb9aa8ae6e516f83f9d4517dedbdeea83f1/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/a4011eb9aa8ae6e516f83f9d4517dedbdeea83f1/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/a4011eb9aa8ae6e516f83f9d4517dedbdeea83f1/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/a4011eb9aa8ae6e516f83f9d4517dedbdeea83f1/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/a4011eb9aa8ae6e516f83f9d4517dedbdeea83f1/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/a4011eb9aa8ae6e516f83f9d4517dedbdeea83f1/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [baker](https://github.com/Minecolonies/minecolonies/blob/a4011eb9aa8ae6e516f83f9d4517dedbdeea83f1/src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker): Add a 'package-info.java' file to document the 'baker' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marchermans** [comment]
@kostronor @raycoms

**Review by xavierh — APPROVED** [review]

**Pull request merged**